src/org/mozilla/javascript/IRFactory.java

```diff
@@ -2466,7 +2466,7 @@ void decompileObjectLiteral(ObjectLiteral node) {
         for (int i = 0; i < size; i++) {
             ObjectProperty prop = props.get(i);
             boolean destructuringShorthand =
-                    Boolean.TRUE.equals(prop.getProp(Node.DESTRUCTURING_SHORTHAND));
+                    Boolean.TRUE.equals(prop.getProp(Node.SHORTHAND_PROPERTY_NAME));
```

Comment:
maybe we need a matching variable name here also

Reply:
5e9aadf Fixed it.